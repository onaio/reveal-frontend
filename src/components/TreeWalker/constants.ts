export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
