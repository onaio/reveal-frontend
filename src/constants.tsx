// strings
export const MAP = 'map';
export type MAP = typeof MAP;
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const ACTIVE_FOCUS_INVESTIGATION = 'Active Focus Investigations';
export type ACTIVE_FOCUS_INVESTIGATION = typeof ACTIVE_FOCUS_INVESTIGATION;
export const FOCUS_AREA_INFO = 'Focus Area Information';
export type FOCUS_AREA_INFO = typeof FOCUS_AREA_INFO;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;
export const MAP_ID = 'map-1';
export type MAP_ID = typeof MAP_ID;
export const GEOJSON = 'geojson';
export type GEOJSON = typeof GEOJSON;
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export type STRINGIFIED_GEOJSON = typeof STRINGIFIED_GEOJSON;
export const FOCUS_AREA_HEADER = 'Focus Area';
export type FOCUS_AREA_HEADER = typeof FOCUS_AREA_HEADER;
export const REASON_HEADER = 'Reason';
export type REASON_HEADER = typeof REASON_HEADER;
export const STATUS_HEADER = 'Status';
export type STATUS_HEADER = typeof STATUS_HEADER;
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export type CASE_NOTIF_DATE_HEADER = typeof CASE_NOTIF_DATE_HEADER;
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export type CASE_CLASSIFICATION_HEADER = typeof CASE_CLASSIFICATION_HEADER;
export const DEFINITIONS = 'Definitions';
export type DEFINITIONS = typeof DEFINITIONS;
export const IN = 'in';
export type IN = typeof IN;
export const DISTRICT = 'District';
export type DISTRICT = typeof DISTRICT;
export const CANTON = 'Canton';
export type CANTON = typeof CANTON;
export const FI_REASON = 'FI Reason';
export type FI_REASON = typeof FI_REASON;
export const FI_STATUS = 'FI Status';
export type FI_STATUS = typeof FI_STATUS;
export const ACTIVE_INVESTIGATION = 'Active Investigation';
export type ACTIVE_INVESTIGATION = typeof ACTIVE_INVESTIGATION;
export const COMPLETE = 'Complete';
export type COMPLETE = typeof COMPLETE;
export const NO = 'No';
export type NO = typeof NO;
export const RESPONSE = 'Response';
export type RESPONSE = typeof RESPONSE;
export const MEASURE = 'Measure';
export type MEASURE = typeof MEASURE;
export const OF = 'of';
export type OF = typeof OF;
export const MARK_AS_COMPLETE = 'Mark as complete';
export type MARK_AS_COMPLETE = typeof MARK_AS_COMPLETE;
export const TARGET = 'Target';
export type TARGET = typeof TARGET;
export const POLYGON = 'Polygon';
export type POLYGON = typeof POLYGON;
export const MULTI_POLYGON = 'MultiPolygon';
export type MULTI_POLYGON = typeof MULTI_POLYGON;
export const POINT = 'Point';
export type POINT = typeof POINT;
export const FEATURE = 'Feature';
export type FEATURE = typeof FEATURE;
export const FEATURE_COLLECTION = 'FeatureCollection';
export type FEATURE_COLLECTION = typeof FEATURE_COLLECTION;
export const MAIN_PLAN = 'main-plan-layer';
export type MAIN_PLAN = typeof MAIN_PLAN;
export const APP = 'APP';
export type APP = typeof APP;
export const NO_GEOMETRIES_RESPONSE = 'Goals have no Geometries';
export type NO_GEOMETRIES_RESPONSE = typeof NO_GEOMETRIES_RESPONSE;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_SINGLE_URL = '/focus-investigation/view';
export type FI_SINGLE_URL = typeof FI_SINGLE_URL;
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export type FI_SINGLE_MAP_URL = typeof FI_SINGLE_MAP_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// task action codes
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export type RACD_REGISTER_FAMILY_CODE = typeof RACD_REGISTER_FAMILY_CODE;
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export type CASE_CONFIRMATION_CODE = typeof CASE_CONFIRMATION_CODE;
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export type MOSQUITO_COLLECTION_CODE = typeof MOSQUITO_COLLECTION_CODE;
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export type BEDNET_DISTRIBUTION_CODE = typeof BEDNET_DISTRIBUTION_CODE;
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export type LARVAL_DIPPING_CODE = typeof LARVAL_DIPPING_CODE;
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export type BLOOD_SCREENING_CODE = typeof BLOOD_SCREENING_CODE;
export const IRS_CODE = 'IRS';
export type IRS = typeof IRS_CODE;
