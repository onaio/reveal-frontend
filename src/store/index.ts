import { store } from '@opensrp/store';

export default store;
