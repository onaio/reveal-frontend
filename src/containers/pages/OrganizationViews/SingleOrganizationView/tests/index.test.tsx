import ListView from '@onaio/list-view';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import Helmet from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { SINGLE_TEAM_URL, TEAM } from '../../../../../constants';
import store from '../../../../../store';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';
import ConnectedSingleOrgView, { SingleOrganizationView } from '../../SingleOrganizationView';

const history = createBrowserHistory();

describe('src/containers/pages/TeamAssignment', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders SingleTeamView without crashing', () => {
    // clean test: check for conditions resulting in errors
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/teamId`,
      },
      team: fixtures.organization1,
      teamMembers: fixtures.organizationMembers,
    };

    shallow(
      <Router history={history}>
        <SingleOrganizationView {...props} />
      </Router>
    );
  });

  it('renders SingleTeamView correctly', () => {
    // clean test: check for stuff that can only be present if page loaded
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/teamId`,
      },
      team: fixtures.organization1,
      teamMembers: [fixtures.organizationMember1],
    };

    const wrapper = mount(
      <Router history={history}>
        <SingleOrganizationView {...props} />
      </Router>
    );

    // check for page title is set
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual(`${TEAM} - User_45`);

    // check for breadcrumbs is set
    expect(toJson(wrapper.find('HeaderBreadcrumb'))).toMatchSnapshot('HeaderBreadcrumb');

    // text wise : check for the team Name text - should be in the top-most tier header
    const pageTitle = wrapper.find('.page-title');
    expect(pageTitle.length).toEqual(1);
    expect(pageTitle.text()).toEqual(`User_45`);

    // check: team details div(snapshot)
    expect(toJson(wrapper.find('#team-details'))).toMatchSnapshot('TeamDetails');

    // check: team members div(snapshot)
    expect(wrapper.find(ListView).length).toEqual(1);
    wrapper.unmount();
  });

  it('Plays well with the store', () => {
    // it gets data from store correctly
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '1' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/1`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleOrgView {...props} />
        </Router>
      </Provider>
    );
    const passedProps = wrapper.find(SingleOrganizationView).props() as any;
    expect(passedProps.team).toEqual(fixtures.organization1);
    expect(passedProps.teamMembers).toEqual([fixtures.organizationMember1]);
  });
});
