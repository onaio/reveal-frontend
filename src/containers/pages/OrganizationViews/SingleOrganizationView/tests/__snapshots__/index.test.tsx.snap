// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/containers/pages/OrganizationViews Deleting api calls for removing a practitioner from an organization: Practitioner 4 removal link 1`] = `
<a
  className="unassign-link text-danger"
  key="healer"
  onClick={[Function]}
>
  Remove
</a>
`;

exports[`src/containers/pages/OrganizationViews renders SingleTeamView correctly: Breadcrumb 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className="reveal-breadcrumb"
  listTag="ol"
  tag="nav"
>
  <nav
    aria-label="breadcrumb"
    className="reveal-breadcrumb"
  >
    <ol
      className="breadcrumb"
    >
      <BreadcrumbItem
        key="0"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="0"
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        key="1"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="1"
            to="/teams"
          >
            <LinkAnchor
              href="/teams"
              navigate={[Function]}
            >
              <a
                href="/teams"
                onClick={[Function]}
              >
                Teams
              </a>
            </LinkAnchor>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        active={true}
        tag="li"
      >
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          The Luang
        </li>
      </BreadcrumbItem>
    </ol>
  </nav>
</Breadcrumb>
`;
